src/window.ts

```diff
@@ -36,4 +37,40 @@ export interface ContainerWindowManager {
      * @returns {ContainerWindow} A new native container window wrapped within a generic ContainerWindow.
      */
     showWindow(url: string, options?: any): ContainerWindow;
+
+    /**
+     * Loads a window layout from persistence
+     * @param {string} name - Name of the window layout to load
+     */
+    loadLayout(name: string): Promise<PersistedWindowLayout>;
+
+    /** Persists a window layout
+     * @param {string} name - Name of the window layout to save
+     * @returns {Promise<PersistedWindowLayout>} - A promise that returns {@link PersistedWindowLayout} if resolved
+     */
+    saveLayout(name: string): Promise<PersistedWindowLayout>;
+
+    /**
+     *  Retrieve a list of persisted window layouts
+     *  @returns {Promise<PersistedWindowLayout[]>} - A promise that returns an array of {@link PersistedWindowLayout} if resolved
+     */
+    getLayouts(): Promise<PersistedWindowLayout[]>;
+}
+
+/** Represents a persisted window in a layout */
+export class PersistedWindow {
+    public name: any;
+    public bounds: any;
+    public url: string;
+    public state?: any;
```

Comment:
@psmulovics Work that is not finalized yet.  Will remove the property from this PR and it will show back up in a future PR around state management.